Formula/qpid-proton.rb

```diff
@@ -0,0 +1,60 @@
+class QpidProton < Formula
+  desc "High-performance, lightweight AMQP 1.0 messaging library."
+  homepage "https://qpid.apache.org/proton/"
+  url "https://www.apache.org/dyn/closer.lua?path=qpid/proton/0.17.0/qpid-proton-0.17.0.tar.gz"
+  sha256 "6ffd26d3d0e495bfdb5d9fefc5349954e6105ea18cc4bb191161d27742c5a01a"
+
+  head do
+    url "git://git.apache.org/qpid-proton.git"
+
+    depends_on "libuv"
+  end
+
+  depends_on "cmake" => :build
+  depends_on "openssl"
+
+  def install
+    args = std_cmake_args
+    # Do not build language bindings
+    args += %w[
+      -DBUILD_CPP=OFF
+      -DBUILD_GO=OFF
+      -DBUILD_JAVASCRIPT=OFF
+      -DBUILD_PERL=OFF
+      -DBUILD_PHP=OFF
+      -DBUILD_PYTHON=OFF
+      -DBUILD_RUBY=OFF
+    ]
+    # Enforce installation into lib/ instead of lib64/
+    args << "-DCMAKE_LIBRARY_OUTPUT_DIRECTORY=#{lib}"
+
+    if MacOS.version == "10.11" && MacOS::Xcode.installed? && MacOS::Xcode.version >= "8.0"
+      inreplace "proton-c/CMakeLists.txt",
+        "CHECK_SYMBOL_EXISTS(clock_gettime \"time.h\" CLOCK_GETTIME_IN_LIBC)",
```

Comment:
This needs to be fixed upstream before we can merge this.

Reply:
@ilovezfs since this is no longer needed in 10.12+, I wonder if it could be removed? We've been working with upstream qpid-proton library on mac for some time now without any problems.

Reply:
I need this line in place to deal with El Capitan which is part of the Homebrew test suite.  This line does not apply for Sierra nor High Sierra.

Reply:
@alanconway Do you think such a patch would make sense in the [upstream Qpid Proton`CMakeLists.txt` file](https://github.com/apache/qpid-proton/blob/master/proton-c/CMakeLists.txt#L139)? Namely, the problem here is that El Capitan has broken so many things and this check for `clock_gettime` is one of them that is required for the Homebrew test to succeed.

Reply:
Yes, I think it can go upstream. We don't really have the resources to test for MacOS but if you have tested it or are confident that somebody has, then I can apply it. Make a JIRA with the patch and assign it to me.

Also FYI: you can now say `cmake -DBUILD_BINDINGS=` to turn all the bindings off or `cmake -DBUILD_BINDINGS=ruby;python` e.g. to build just ruby and python

Reply:
@alanconway Is that `cmake -DBUILD_BINDINGS=` in 0.17.0?